indexer/ai/src/main/java/com/google/cloudsearch/ai/StandardSkillCategoryExtraction.java

```diff
@@ -170,11 +173,24 @@ public void executeSkill(String contentOrURI, Multimap<String, Object> structure
                     }
                 }
             }
-            language.shutdown();
-            language.awaitTermination(30, TimeUnit.SECONDS);
-            language.close();
-        } catch (IOException | InterruptedException e) {
-            e.printStackTrace();
+        } catch (IOException e) {
+            log.error(e);
+        }
+    }
+
+    /**
+     * Shutdown LanguageService Client
+     */
+    @Override
+    public void shutdownSkill() {
```

Comment:
Who will call shutdownSkill? The function where skill gets initialized?

Reply:
No, it will be called by the driver just after execute:
So:
skill.execute()
skill.shutdown()